python/box.tests/test/org/jetbrains/kotlin/python/test/BasicIrBoxTest.kt

```diff
@@ -138,23 +142,37 @@ abstract class BasicIrBoxTest(
             }
 
             if (!skipRegularMode) {
-                val compiledModule = compile(
-                    project = config.project,
-                    mainModule = MainModule.SourceFiles(filesToCompile),
-                    analyzer = AnalyzerWithCompilerReport(config.configuration),
-                    configuration = config.configuration,
-                    phaseConfig = phaseConfig,
-                    irFactory = IrFactoryImpl,
-                    allDependencies = resolvedLibraries,
-                    friendDependencies = emptyList(),
-                    mainArguments = mainCallParameters.run { if (shouldBeGenerated()) arguments() else null },
-                    exportedDeclarations = setOf(FqName.fromSegments(listOfNotNull(testPackage, testFunction))),
-                    generateFullJs = true,
-                    generateDceJs = runIrDce,
-                    es6mode = runEs6Mode,
-                    multiModule = splitPerModule || perModule,
-                    propertyLazyInitialization = propertyLazyInitialization,
-                )
+                var compilationException: KotlinCompilationException? = null
+                val compiledModuleWithDuration: Pair<Long, CompilerResult?> = measureTimeMillisWithResult {
+                    try {
+                        compile(
```

Comment:
We have another `compile` call 40 lines below. Should we surround it too?

Reply:
From my empirical tests I found out that the other `compile` doesn't execute in our case (i.e. we have some dead, leftover code from the JS backend). But I'll double-check :+1:

Reply:
At least I have `out-pir` dir that seems to be `out` but with DCE applied. So maybe that `compile` is executed but its output isn't used to run the interpreter.

Maybe we don't need it for now, but it would be cool to have DCE in the future, so I wouldn't say it's redundant code. However, if it's not used now, feel free to don't spend time on it for now.

Reply:
Yeah, let's go forward with the change as is, and when I work on generating the new report, I'm sure I'll see if some tests are missing information added in this PR, then I'll fix it iteratively.